packages/models/src/cache/onlineStatus.ts

```diff
@@ -0,0 +1,31 @@
+import { CacheBase } from "./base";
+
+export class OnlineStatus extends CacheBase {
+  readonly key = "online-status";
+
+  async getAll(): Promise<Record<string, string>> {
+    return await this.client.hGetAll(this.key)
+  }
+
+  async addUser(userId: number): Promise<void> {
+    await this.client.hSet(this.key, this.asField(userId), this.encode(1));
+  }
+
+  async removeUser(userId: number): Promise<void> {
+    await this.client.hDel(this.key, this.asField(userId));
+  }
+
+  async isOnline(userId: number): Promise<boolean> {
+    return await this.client.hExists(this.key, this.asField(userId));
+  }
+
+  async isOnlineBatch(userIds: number[]): Promise<boolean[]> {
+    return await Promise.all(
+      userIds.map(async id => this.client.hExists(this.key, this.asField(id)))
+    );
+  }
```

Comment:
I don't know if it will work or not but can you try to use `.multi()` with `.exec()`. 

```ts
const builder = this.client.multi();

userIds.forEach((id) => {
    builder.exists(this.key, this.asField(id));
})

const result = await builder.exec(); // result ?? 
```